user/views.py

```diff
@@ -1,3 +1,67 @@
-from django.shortcuts import render
+import os
+import json
+import bcrypt
+import re
+import jwt
+import my_settings
+import utils
+
+from django.views           import View
+from user.models            import User
+from django.http            import JsonResponse
 
 # Create your views here.
+class EmailSignUpView(View):
+    def post(self, request):
+        try:
+            data  = json.loads(request.body)
+            regex = '^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$'
+            if not (re.search(regex, data['email'])):
+                return JsonResponse({"message": "INVALID_EMAIL"}, status=400)
+
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({"message": "EMAIL_EXISTS"}, status=400)
+
+            password       = data['password'].encode('utf-8')
+            password_crypt = bcrypt.hashpw(password, bcrypt.gensalt()).decode('utf-8')
+
+            User(
+                name                = data['name'],
+                password            = password_crypt,
+                email               = data['email'],
+                location_is_agreed  = data['location_is_agreed'],
+                promotion_is_agreed = data['promotion_is_agreed'],
+            ).save()
+
+            return JsonResponse({"message": "SIGNUP_SUCCESS"}, status=200)
+
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+
+        except ValueError:
+            return JsonResponse({"message": "VALUE_ERROR"}, status=400)
+
+
+class EmailSignInView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            user = User.objects.get(email=data['email'])
+
+            if bcrypt.checkpw(data['password'].encode('UTF-8'), user.password.encode('UTF-8')):
+                key       = my_settings.SECRET.get('SECRET_KEY')
+                algorithm = my_settings.SECRET.get('JWT_ALGORITHM')
+                token     = jwt.encode({'user' : user.id},key, algorithm = algorithm).decode('UTF-8')
+                return JsonResponse({"token": token, "message": "SIGNIN_SUCCESS", "name" : user.name}, status=200)
+
+            else:
+                return JsonResponse({"message": "INVALID_USER"}, status=401)
```

Comment:
```suggestion
                return JsonResponse({"message": "INVALID_PASSWORD"}, status=401)
```
가 맞는데 그냥 식별 안되게 하려고 같은 에러메세지로 한건가요? 그런거라면 그냥 메세지를 없애는게 더 좋은 방법입니다.

Reply:
네 이메일이 틀린경우와 비밀번호 틀린 경우가 식별되지 않게 하기 위해 같은 에러메시지를 넣었습니다. 메시지 수정하였습니다:)